internal/service/advancedclustertpf/resource.go

```diff
@@ -112,25 +113,35 @@ func (r *rs) Update(ctx context.Context, req resource.UpdateRequest, resp *resou
 		diags.AddError("errorPatchPayload", err.Error())
 		return
 	}
-	upgradeRequest := getTenantUpgradeRequest(stateReq, patchReq)
 	var cluster *admin.ClusterDescription20240805
-	if upgradeRequest != nil {
-		cluster = r.applyTenantUpgrade(ctx, &plan, upgradeRequest, diags)
-	} else {
-		cluster = r.applyClusterChanges(ctx, diags, &state, &plan, patchReq)
-	}
-	if diags.HasError() {
-		return
+	if !update.IsZeroValues(patchReq) {
+		upgradeRequest := getTenantUpgradeRequest(stateReq, patchReq)
+		if upgradeRequest != nil {
+			cluster = r.applyTenantUpgrade(ctx, &plan, upgradeRequest, diags)
+		} else {
+			cluster = r.applyClusterChanges(ctx, diags, &state, &plan, patchReq)
+		}
+		if diags.HasError() {
+			return
+		}
 	}
-	legacyAdvConfig, advConfig := r.applyAdvancedConfigurationChanges(ctx, diags, &state, &plan)
+	legacyAdvConfig, advConfig, advConfigChanged := r.applyAdvancedConfigurationChanges(ctx, diags, &state, &plan)
 	if diags.HasError() {
 		return
 	}
 	var model *TFModel
-	if cluster == nil { // no cluster updates, TODO: See if we can only update the advanced config if no cluster changes
```

Comment:
Only updating the **changed fields** doesn't work, we will get errors similar to:
```

  | Error: Value Conversion Error
  | 
  |   with mongodbatlas_advanced_cluster.test,
  |   on terraform_plugin_test.tf line 13, in resource "mongodbatlas_advanced_cluster" "test":
  |   13: \t\t\ttimeouts = {
  |   14: \t\t\t\tcreate = "20s"
  |   15: \t\t\t}
  | 
  | An unexpected error was encountered while verifying an attribute value
  | matched its expected type to prevent unexpected behavior or panics. This is
  | always an error in the provider. Please report the following to the provider
  | developer:
  | 
  | Expected framework type from provider logic: timeouts.Type / underlying type:
  | tftypes.Object["create":tftypes.String, "delete":tftypes.String,
  | "update":tftypes.String]
  | Received framework type from provider logic: timeouts.Type / underlying type:
  | tftypes.Object[]
  | Path: timeouts
  ```
  However, we can workaround this by using the values from the `state`.
  In this PR we support using state values when:
  1. No changes to advanced_configuration, use value from state instead of re-reading
     1. This should be safe since a read of advanced_configuration is anyway done at the planning stage, and if no changes are detected we can re-use state value and save two API calls
  2. No changes to cluster, only update advanced config, see updateAdvConfig method